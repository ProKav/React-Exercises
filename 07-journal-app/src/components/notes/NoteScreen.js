import React from 'react'
import { NotesAppBar } from './NotesAppBar'

export const NoteScreen = () => {
	return (
		<div className='notes__main-content'>
			<NotesAppBar />
			<div className='notes__content'>
				<input
					type='text'
					placeholder='Some awesome title'
					className='notes__title-input'
					autoComplete='off'
				/>
				<textarea
					placeholder='Some to remaind today?'
					className='notes__textarea'></textarea>
				<div className='notes__image'>
					<img
						src='http://unsplash.it/g/300?random&gravity=center'
						alt='some kitties'
					/>
				</div>
			</div>
		</div>
	)
}
