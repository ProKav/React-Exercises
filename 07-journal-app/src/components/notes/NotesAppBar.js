import React from 'react'
import { useDispatch } from 'react-redux'
import { startSaveNote } from '../../actions/notes'
import { useSelector } from 'react-redux'

export const NotesAppBar = () => {
	const dispatch = useDispatch()
	const { active } = useSelector(state => state.notes)

	const handleSave = () => {
		dispatch(startSaveNote(active))
	}

	return (
		<div className='notes__appbar'>
			<span>17 de Julio de 2021</span>
			<div>
				<button className='btn'>Picture</button>
				<button className='btn' onClick={handleSave}>
					Save
				</button>
			</div>
		</div>
	)
}
