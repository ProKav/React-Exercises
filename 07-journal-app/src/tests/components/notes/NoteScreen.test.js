import React from 'react'
import { mount } from 'enzyme'
import { Provider } from 'react-redux'

import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import '@testing-library/jest-dom'
import { NoteScreen } from '../../../components/notes/NoteScreen'
import { activeNote } from '../../../actions/notes'

jest.mock('../../../actions/notes', () => ({
	activeNote: jest.fn(),
}))

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initState = {
	auth: {
		uid: '1',
		name: 'Kamilo',
	},
	ui: {
		loading: false,
		msgError: null,
	},
	notes: {
		active: {
			id: 1234,
			title: 'Holi',
			body: 'Mundillo',
			date: 0,
		},
		notes: [],
	},
}

let store = mockStore(initState)
store.dispatch = jest.fn()

const wrapper = mount(
	<Provider store={store}>
		<NoteScreen />
	</Provider>,
)

describe('Test in <NoteScreen /> component', () => {
	test('Should be display correctly', () => {
		expect(wrapper).toMatchSnapshot()
	})

	test('Should dispatch activeNote', () => {
		wrapper.find('input[name="title"]').simulate('change', {
			target: {
				name: 'title',
				value: 'Hello Again',
			},
		})

		expect(activeNote).toHaveBeenLastCalledWith(1234, {
			body: 'Mundillo',
			title: 'Hello Again',
			id: 1234,
			date: 0,
		})
	})
})
