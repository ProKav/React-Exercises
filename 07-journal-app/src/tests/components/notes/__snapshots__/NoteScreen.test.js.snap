// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <NoteScreen /> component Should be display correctly 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      17 de Julio de 2021
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awesome title"
      type="text"
      value="Holi"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="Some to remaind today?"
      value="Mundillo"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
